.pre-commit-config.yaml

```diff
@@ -23,7 +31,7 @@ repos:
     rev: v1.3.5
     hooks:
       - id: clang-format
-        args: ["--style=file"]  # Use the .clang-format file for configuration
+        args: ["--style=file", "-i"]  # Use the .clang-format file for configuration and apply all fixes
```

Comment:
Wait, why do we need "-i"? We will need pre-commit to report unformatted code, and with this flag clang-format will not report error on the CI.

Reply:
It should still give errors but for now local pre-commit runs also fixes formatting issues.

Reply:
Can you confirm this? Maybe I remember wrong.

Reply:
@tigercosmos https://github.com/seladb/PcapPlusPlus/actions/runs/10502860257/job/29095177138?pr=1543

Reply:
so why do we need the `-i` argument?

Reply:
So we a user run pre-commit, the wrong format will be fixed automatically. Currently we need to manually run clang-format.

Reply:
oh ok 👍